lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -180,6 +191,11 @@ public void OnNext(ICloseEvent closeEvent)
         /// </summary>
         public void Dispose()
         {
+            if (Interlocked.Read(ref _disposed) == 0)
```

Comment:
This does not work, you need to use the result of `Interlocked.Exchange` instead.